### ISSUE TITLE
[JIT] LSTMCell still not speedup with 1.1.0

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#20060]
I've tried the [template provided in fastrnn](https://github.com/pytorch/benchmark/blob/master/rnns/fastrnns/custom_lstms.py#L94-L120) with latest PyTorch, however, it seems the speedup still not there yet. Here is the benchmark code

```python
from torch import jit
import torch
import torch.nn as nn
import torch.nn.functional as F

torch.manual_seed(0)

class LSTMCell(jit.ScriptModule):
    def __init__(self, input_size, hidden_size):
        super(LSTMCell, self).__init__()
        self.input_size = input_size
        self.hidden_size = hidden_size
        self.weight_ih = nn.Parameter(torch.randn(4 * hidden_size, input_size))
        self.weight_hh = nn.Parameter(torch.randn(4 * hidden_size, hidden_size))
        self.bias_ih = nn.Parameter(torch.randn(4 * hidden_size))
        self.bias_hh = nn.Parameter(torch.randn(4 * hidden_size))

    @jit.script_method
    def forward(self, input, state):
        # type: (Tensor, Tuple[Tensor, Tensor]) -> Tuple[Tensor, Tuple[Tensor, Tensor]]
        hx, cx = state
        gates = (torch.mm(input, self.weight_ih.t()) + self.bias_ih +
                 torch.mm(hx, self.weight_hh.t()) + self.bias_hh)
        ingate, forgetgate, cellgate, outgate = gates.chunk(4, 1)

        ingate = torch.sigmoid(ingate)
        forgetgate = torch.sigmoid(forgetgate)
        cellgate = torch.tanh(cellgate)
        outgate = torch.sigmoid(outgate)

        cy = (forgetgate * cx) + (ingate * cellgate)
        hy = outgate * torch.tanh(cy)

        return hy, (hy, cy)
    
import time

x = torch.randn(64, 512)
h = torch.zeros(64, 512)
c = torch.zeros_like(h)

rnn1 = LSTMCell(512, 512)
rnn2 = nn.LSTMCell(512, 512)

t = time.perf_counter()
rnn1(x, (h, c))
print(time.perf_counter() - t)

t = time.perf_counter()
rnn2(x, (h, c))
print(time.perf_counter() - t)

>>> 0.06589111499488354
>>> 0.006260981783270836
```

```
 - PyTorch Version: 1.1.0
 - OS: Ubuntu 18.04
 - How you installed PyTorch: pip
 - Python version: 3.7
```

**Comment by SsnL** [comment]
fwiw running this on a master build with some warmup iters gives me numbers like
```
0.0015532950055785477
0.001382627000566572
```

**Comment by wanchaol** [comment]
@zuoxingdong are you running this on CPU? the first time when you pass inputs to the JIT code, it will run optimization passes to compile the specialization for that input shape, and then generate optimized graph. So in your case, as @SsnL suggested, you will need to have a small warmup before you time it. You can also try to time it on GPU, which should result in similar times between `LSTMCell` and `nn.LSTMCell`

**Comment by wanchaol** [comment]
close this since it should have be resolved. @zuoxingdong if you feel any difficulties in reproducing a comparable numbers, feel free to reopen it :)
